### PR TITLE
Updated accessList for judging pages

### PR DESCRIPTION
**Pull request by DavidWilliamK**
Removed deprecated routes from access list
Added new routes

### CODE REVIEW

File: function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            id = "addNewUrlRegexForJudgingOperations",
+            order = "0003")
+    public void addUrlRegexForJudgingOperations(
+            MongoTemplate mongoTemplate
+    ) {
+
+        String urlRegex = "^\\/batches\\/[A-Za-z0-9\\-]+\\/final-judging\\/(add|report-page|(.*\\/(detail|edit)))(\\/)?$";
+
+        Map<String, Object> adminComponents = new HashMap<>();
+        adminComponents.put("add", true);
```
Comment (oliverrsebastian) [inline function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: sure about allowing admin to add final score in final judging detail?

File: function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        adminComponents.put("read", true);
+
+        Map<String, Object> studentComponents = new HashMap<>();
+        studentComponents.put("add", false);
+        studentComponents.put("delete", false);
+        studentComponents.put("edit", false);
+        studentComponents.put("read", false);
+
+        Map<String, Object> otherComponents = new HashMap<>();
+        otherComponents.put("add", false);
```
Comment (oliverrsebastian) [inline function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: what about judge authority to give final score in final judging detail?

File: function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+public class DataMigration_020 {
+
+    @ChangeSet(author = "david",
+            id = "addUrlRegexForJudgingList",
+            order = "0001")
+    public void addUrlRegexForJudgingList(
+            MongoTemplate mongoTemplate
+    ) {
+
+        String urlRegex = "^\\/final-judging(\\/)?$";
```
Comment (jonathan016) [inline function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: Need to include query params? If so maybe `urlRegex` can be changed to `^\\/final-judging(\\/)?(\\?.*)?$`

Comment (jonathan016) [reply function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: cc @DavidWilliamK

File: function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    public void removeDeprecatedJudgingRegex(
+            MongoTemplate mongoTemplate
+    ) {
+        Criteria criteria = Criteria.where(FieldName.Access.URL_REGEX)
+                .is("^\\/batches\\/[A-Za-z0-9\\-]+\\/(assignments|quizzes|final-judging)(\\/|\\/[A-Za-z0-9\\-]+)?(\\/|\\/(detail|addDetail|add)(\\/)?)?$");
+        List<Access> accesses = mongoTemplate.find(
+                Query.query(criteria), Access.class);
+
+        accesses.forEach(access -> {
+            access.setUrlRegex("^\\/batches\\/[A-Za-z0-9\\-]+\\/(assignments|quizzes)(\\/|\\/[A-Za-z0-9\\-]+)?(\\/|\\/(detail|addDetail|add)(\\/)?)?$");
```
Comment (jonathan016) [inline function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: Need to include query params? If so maybe `urlRegex` can be appended with `(\\?.*)?` prior to `$` sign

Comment (jonathan016) [reply function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: cc @DavidWilliamK

File: function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+    @ChangeSet(author = "david",
+            id = "addNewUrlRegexForJudgingOperations",
+            order = "0003")
+    public void addUrlRegexForJudgingOperations(
+            MongoTemplate mongoTemplate
+    ) {
+
+        String urlRegex = "^\\/batches\\/[A-Za-z0-9\\-]+\\/final-judging\\/(add|report-page|(.*\\/(detail|edit)))(\\/)?$";
```
Comment (jonathan016) [inline function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: Need to include query params? If so maybe `urlRegex` can be appended with `(\\?.*)?` prior to `$` sign

cc @DavidWilliamK

File: function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        Access studentAccess = Access.builder()
+                .urlRegex(urlRegex)
+                .components(studentComponents)
+                .role(Role.STUDENT)
+                .build();
+
+        mongoTemplate.insert(studentAccess, DocumentName.ACCESS);
+    }
+
+
```
Comment (jonathan016) [inline function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: Remove extra space

File: function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    @ChangeSet(author = "david",
+            id = "removeDeprecatedJudgingRegex",
+            order = "0002")
+    public void removeDeprecatedJudgingRegex(
+            MongoTemplate mongoTemplate
+    ) {
+        Criteria criteria = Criteria.where(FieldName.Access.URL_REGEX)
```
Comment (jonathan016) [inline function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: Add extra space before line

File: function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        Access studentAccess = Access.builder()
+                .urlRegex(urlRegex)
+                .components(studentComponents)
+                .role(Role.STUDENT)
+                .build();
+
+        mongoTemplate.insert(studentAccess, DocumentName.ACCESS);
+    }
+
+}
```
Comment (jonathan016) [inline function-data-migration/src/main/java/com/future/function/data/migration/change/log/DataMigration_020.java]: Add extra line after

**Pull request closed without merging**